engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/Graph.scala

```diff
@@ -24,11 +27,26 @@ sealed class Graph extends Serializable {
     val copy = new Graph
     copy.rootScope     = this.rootScope.deepCopy(scope_mapping)
     copy.links         = this.links
+    copy.sourceLinks   = this.sourceLinks
+    copy.targetLinks   = this.targetLinks
     copy.globalSymbols = this.globalSymbols
     copy.nextIdCounter = this.nextIdCounter
     copy
   }
 
+  def initLinks(links: Set[Graph.Link]): Unit = {
+    links.foreach { link =>
+      sourceLinks = sourceLinks.updatedWith(link.source)(v =>
+        v.map(s => s + link).orElse(Some(Set(link)))
+      )
+      targetLinks = targetLinks.updatedWith(link.target)(v =>
+        v.map(s => s + link).orElse(Some(Set(link)))
+      )
+    }
+  }
+
+  def getLinks(): Set[Graph.Link] = links
```

Comment:
Wouldn't it be better to remove the `links` field all together and just compute the links from `sourceLinks` and `targetLinks`? The logic between `initLinks` and `getLinks` & co. would be simpler and possibly `getLinks()` isn't on critical path.

Reply:
I kind of like that `links` is the source of truth and `sourceLinks/targetLinks` serve as aux indexes for quicker access. It also simplifies serde.